draft-ietf-httpbis-http2bis.xml

```diff
@@ -1360,13 +1363,11 @@
         <figure anchor="DATAFramePayload">
           <name>DATA Frame Payload</name>
           <artwork type="inline"><![CDATA[
- +---------------+
- |Pad Length? (8)|
- +---------------+-----------------------------------------------+
- |                            Data (*)                         ...
- +---------------------------------------------------------------+
- |                           Padding (*)                       ...
- +---------------------------------------------------------------+
+Data Frame Payload {
+  [Pad Length (8)],
+  Data (..),
+  Padding (..),
+}
```

Comment:
If we're going full QUIC, then I think we should include the frame header in each of these frame definitions, just like QUIC ended up doing.  The advantage of that being that you can put the flag bits into the definition.  Then we can unify the descriptions of flags and fields.

Sure, that's a lot more work, but I think that it's worth doing.

Reply:
So, something like 

```
Data Frame {
  Length (24),
  Type (8),
  END_STREAM (1),
  Flag 1 (1),
  Flag 2 (1),
  PADDED (1),
  Flag 4 (1),
  Flag 5 (1),
  Flag 6 (1),
  Flag 7 (1),
  R (1),
  Stream Identifier (31),
  [Pad Length (8)],
  Data (..),
  Padding (..),
} 
```

?

Reply:
I would probably compress the unused flags, but yeah.

```  
  END_STREAM (1),
  Unused Flags (2),
  PADDED (1),
  Unused Flags (4),
```

Or maybe
```
  END_STREAM Flag (1),
```

The main problem with this structure is that this probably needs to come with changes to the way that these are described.

Reply:
Understood re: descriptions; just want to get the format convention down first.

I didn't compress unused flags because it seems like it'd be super-useful for subsequent extensions that want to define flags. Thoughts?

Reply:
![](https://media.giphy.com/media/y65VoOlimZaus/giphy.gif)

Reply:
Now I'm wondering whether it makes sense to leave flags in a separate prose list, or move them up into the same list as the other descriptions. Got any gifs for that?